include/tut/tut_assert.hpp

```diff
@@ -130,12 +130,16 @@ void ensure_equals(const M& msg, const LHS * const actual, const RHS * const exp
     }
 }
 
-template<typename M>
-void ensure_equals(const M& msg, const double& actual, const double& expected, const double& epsilon)
+template<typename M, typename T>
+typename std::enable_if<std::is_floating_point<T>::value, void>::type
+    ensure_equals(const M& msg, const T& actual, const T& expected, const T& epsilon)
 {
-    const double diff = actual - expected;
-
-    if ( (actual != expected) && !((diff <= epsilon) && (diff >= -epsilon )) )
+	if (std::isnan(actual) && std::isnan(expected))
+		return;
```

Comment:
Usually, `NaN != NaN`, but in a test library it may be better to have `NaN == NaN`, what do you think?

I have the case where I want to assert that a value is `NaN`, so I have 2 options:

```c++
ensure_not(myNaN == myNaN); // current version
ensure_equals(myNaN, quiet_NaN(); // if we use the suggested code
```

Reply:
I'd rather have `ensure_isnan(value)` instead of abusing comparison rules.

It might even be a good idea to add generic `ensure_call(<function>, <arguments...>)`, provided we're using a recent standard.